### PR TITLE
Updated forgetful settings option label

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/33048

image

Resolves 

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/android/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -723,7 +723,7 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
         Don't upgrade connections to HTTPS (disabled)
       </message>
       <message name="IDS_BRAVE_FORGET_FIRST_PARTY_STORAGE_SWITCH" desc="Text for First Party Storage cleanup shields setting.">
-        Forget me when I close this site
+        Forget me when I close every site, by default
```
Comment (mkarolin) [inline browser/ui/android/strings/android_brave_strings.grd:726]: Defer to @rmcfadden3 here.

Comment (mkarolin) [reply browser/ui/android/strings/android_brave_strings.grd:726]: Considering that this is a toggle "name" that's followed by the explanation `Clears cookies and other site data when you close a site`, I would just go for somethings like `Make sites forget me`, but again, defer to @rmcfadden3 here.

Comment (rmcfadden3) [reply browser/ui/android/strings/android_brave_strings.grd:726]: Yeah, we don't want to imply it's Brave doing the forgetting. Also, I notice the other settings don't use pronouns like "I" or "me" in them. So I wonder if we should rather do:

How about `Clear session data when a site is closed`

Comment (simonhong) [reply browser/ui/android/strings/android_brave_strings.grd]: Updated and changed the screenshot.

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**